app/scripts/lib/multichain-api/wallet-getPermissions.test.js

```diff
@@ -0,0 +1,282 @@
+        id: '3',
+        parentCapability: 'otherPermission',
+        caveats: [
+          {
+            value: {
+              foo: 'bar',
+            },
+          },
+        ],
+      },
+    }),
+  );
+  const response = {};
+  const handler = (request) =>
+    getPermissionsHandler.implementation(request, response, next, end, {
+      getPermissionsForOrigin,
+    });
+
+  return {
+    response,
+    next,
+    end,
+    getPermissionsForOrigin,
+    handler,
+  };
+};
+
+describe('getPermissionsHandler', () => {
+  beforeAll(() => {
+    delete process.env.BARAD_DUR;
+  });
+
+  it('gets the permissions for the origin', () => {
+    const { handler, getPermissionsForOrigin } = createMockedHandler();
+
+    handler(baseRequest);
+    expect(getPermissionsForOrigin).toHaveBeenCalled();
+  });
+
+  it('does not return the CAIP-25 endowment', () => {
+    const { handler, response } = createMockedHandler();
+
+    handler(baseRequest);
+    expect(response.result).toStrictEqual([
+      {
+        id: '1',
+        parentCapability: 'eth_accounts',
+        caveats: [
+          {
+            value: ['0xdead', '0xbeef'],
+          },
+        ],
+      },
+      {
+        id: '3',
+        parentCapability: 'otherPermission',
+        caveats: [
+          {
+            value: {
+              foo: 'bar',
+            },
+          },
+        ],
+      },
+    ]);
+  });
+
+  describe('BARAD_DUR flag is set', () => {
+    beforeAll(() => {
+      process.env.BARAD_DUR = 1;
+    });
+
+    it('returns the permissions without the CAIP-25 endowment if the CAIP-25 caveat is not found', () => {
+      const { handler, getPermissionsForOrigin, response } =
+        createMockedHandler();
+
+      getPermissionsForOrigin.mockReturnValue(
+        Object.freeze({
+          eth_accounts: {
+            id: '1',
+            parentCapability: 'eth_accounts',
+            caveats: [
+              {
+                value: ['0xdead', '0xbeef'],
+              },
+            ],
+          },
+          otherPermission: {
+            id: '2',
+            parentCapability: 'otherPermission',
+            caveats: [
+              {
+                value: {
+                  foo: 'bar',
+                },
+              },
+            ],
+          },
+        }),
+      );
+
+      handler(baseRequest);
+      expect(response.result).toStrictEqual([
+        {
+          id: '1',
+          parentCapability: 'eth_accounts',
+          caveats: [
+            {
+              value: ['0xdead', '0xbeef'],
+            },
+          ],
+        },
+        {
+          id: '2',
+          parentCapability: 'otherPermission',
+          caveats: [
+            {
+              value: {
+                foo: 'bar',
+              },
+            },
+          ],
+        },
+      ]);
+    });
+
+    it('returns the permissions without eth_accounts and the CAIP-25 endowement if there are no accounts authorized for eip155 namespaces', () => {
```

Comment:
🤔 this actually seems kindof like a strange thing to do. We enforce that permissions we recognize contain authorized accounts, but then just expose totally random permissions we don't recognize?

Reply:
See: https://github.com/MetaMask/metamask-extension/pull/26054#discussion_r1692154398